### PR TITLE
Issue 73

### LINKED ISSUE #73: Cache key must be based on a version number of the source index
Currently the cache key is generated from the join query definition. However, the data in the index can change and two identical queries can return different result sets. 
Given that the cache key is used in the lucene query cache and in the siren join cache, this can lead to incorrect results.
Instead, we have to create a cache key that will be dependent on the version of the source index.

### PR DESCRIPTION
**Pull request by rendel**
closes #73

### CODE REVIEW

File: src/main/java/solutions/siren/join/action/coordinate/CoordinateSearchResponse.java [hunk 1]
```diff
@@ -28,6 +28,7 @@
 import org.elasticsearch.search.SearchHits;
 import org.elasticsearch.search.aggregations.Aggregations;
 import org.elasticsearch.search.suggest.Suggest;
+import solutions.siren.join.action.coordinate.execution.CoordinateSearchMetadata;
```
Comment (scampi) [inline src/main/java/solutions/siren/join/action/coordinate/CoordinateSearchResponse.java]: seems to be unused

File: src/main/java/solutions/siren/join/action/coordinate/execution/FilterJoinVisitor.java [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+   * has been removed.
+   */
+  private boolean removeConvertedNodes(AbstractNode node) {
+    boolean nodeRemoved = false;
+    Iterator<AbstractNode> it = node.getChildren().iterator();
+    while (it.hasNext()) {
+      FilterJoinNode child = (FilterJoinNode) it.next();
+      if (child.getState().equals(FilterJoinNode.State.CONVERTED)) {
+        it.remove();
+        nodeRemoved |= true;
```
Comment (scampi) [inline src/main/java/solutions/siren/join/action/coordinate/execution/FilterJoinVisitor.java]: is the bitwise intentional ? seems ok though http://stackoverflow.com/a/1724263

Comment (rendel) [inline src/main/java/solutions/siren/join/action/coordinate/execution/FilterJoinVisitor.java]: yes

**Comment by scampi** [comment]
👍

**Comment by rendel** [comment]
thanks

**Pull request merged**